### PR TITLE
Finalize javascript_nodejs/05.multi-turn-prompt

### PR DESCRIPTION
**Pull request by benbrown**
- [x] Is numbered and named correctly and following the spec
- [x] All files have been spell checked
- [x] All style cop/ linting issues have been addressed
- [x] Include all DCR changes
- [x] Explicit state save
- [x] Method name changes begin, end etc
- [x] Does not use mainDialog unless that file actually extends Dialog/ ComponentDialog
- [x] Compiles with latest packages
- [x] Runs per instructions in README
- [x] Add commented out code to use blob storage
- [x] Update to look for production/ local (bot configuration)
- [x] onTurnError at adapter
- [x] Include .env or appsettings.json or web.config with botFilePath and botFileSecret
- [x] Deployment scripts folder has been cleaned up and only includes msbot export files
- [x] README to include
   - [x] msbot clone steps
   - [x] Tools install/ uninstall
   - [x] Concise set of steps to run the sample
- [x] ~C# sample -- Include .bot file in the build output~
- [x] All samples include the latest botfile (no "secretKey" field; replaced with "padLock"). Use latest MSBOT off the tools MyGet feed to generate the bot file
- [x] remove and sort usings
- [x] ~remove azuredeploy.* under deploymentscripts folder (if any)~

### CODE REVIEW

File: samples/javascript_nodejs/05.multi-turn-prompt/bot.js [hunk 1]
```diff
@@ -165,4 +165,4 @@ class MainDialog {
     }
 }
 
-module.exports = MainDialog;
+module.exports = MultiTurnBot;
```
Comment (sgellock) [inline samples/javascript_nodejs/05.multi-turn-prompt/bot.js]: stevenic has suggested that we want to always to named exports.  so module.exports.MultiTurnBot = MultiTurnBot.  He likes this pattern as it forces the naming of the associated require();

File: samples/javascript_nodejs/05.multi-turn-prompt/index.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ const CONFIG_ERROR = 1;
 const { BotFrameworkAdapter, ConversationState, MemoryStorage, UserState } = require('botbuilder');
 const { BotConfiguration } = require('botframework-config');
 
-const MainDialog = require('./dialogs/mainDialog');
+const MultiTurnBot = require('./bot');
```
Comment (sgellock) [inline samples/javascript_nodejs/05.multi-turn-prompt/index.js]: Change to 
const { MultiTurnBot } = require('./bot');

File: samples/javascript_nodejs/05.multi-turn-prompt/multi-turn-prompt.bot [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {
-    "name": "multiturn-prompts-bot",
-    "secretKey": "",
+    "name": "multi-turn-prompt",
+    "padLock": "",
     "services": [
         {
             "type": "endpoint",
-            "name": "multiturn-prompts-bot",
+            "name": "multi-turn-prompt",
```
Comment (sgellock) [inline samples/javascript_nodejs/05.multi-turn-prompt/multi-turn-prompt.bot]: We want to have all local services use the "name": "development" pattern.  in services that have a local and an azure/Luis set of services, this gives us the ability to have local and cloud end points.

File: samples/javascript_nodejs/05.multi-turn-prompt/index.js [hunk 4]
```diff
@@ -40,7 +40,7 @@ try {
 
 // Define the name of the bot, as specified in .bot file.
 // See https://aka.ms/about-bot-file to learn more about .bot files.
-const BOT_CONFIGURATION = 'multiturn-prompts-bot';
+const BOT_CONFIGURATION = 'multi-turn-prompt';
```
Comment (sgellock) [inline samples/javascript_nodejs/05.multi-turn-prompt/index.js]: this should change to:
const BOT_CONFIGURATION = 'development';

File: samples/javascript_nodejs/05.multi-turn-prompt/.env [hunk 5]
```diff
@@ -0,0 +1 @@
+botFilePath=./multi-turn-prompt.bot
```
[the remark this answers is not in the document]
Comment (benbrown) [reply samples/javascript_nodejs/05.multi-turn-prompt/.env]: will do

File: samples/javascript_nodejs/05.multi-turn-prompt/bot.js [hunk 6]
```diff
@@ -133,15 +133,15 @@ class MainDialog {
...
                 const user = await this.userProfile.get(dc.context, {});
                 if (user.name) {
-                    await dc.beginDialog(HELLO_USER)
+                    await dc.beginDialog(HELLO_USER);
                 } else {
-                    await dc.beginDialog(WHO_ARE_YOU)
+                    await dc.beginDialog(WHO_ARE_YOU);
                 }
             }
         } else if (
```
[the remark this answers is not in the document]
Comment (benbrown) [reply samples/javascript_nodejs/05.multi-turn-prompt/bot.js]: What part of this don't you like?

Do you prefer this test be in a dialog step function?

[the remark this answers is not in the document]
Comment (benbrown) [reply samples/javascript_nodejs/05.multi-turn-prompt/bot.js]: Fixed!

File: samples/javascript_nodejs/05.multi-turn-prompt/multi-turn-prompt.bot [hunk 7]
```diff
@@ -1,11 +1,11 @@
 {
-    "name": "multiturn-prompts-bot",
-    "secretKey": "",
+    "name": "multi-turn-prompt",
+    "padlock": "",
```
Comment (vishwacsena) [inline samples/javascript_nodejs/05.multi-turn-prompt/multi-turn-prompt.bot]: Also missing ,
  "version": "2.0". Here's a latest .bot file sample -- 

{
  "name": "Welcome Users Sample",
  "services": [
    {
      "type": "endpoint",
      "name": "development",
      "endpoint": "http://localhost:3978/api/messages",
      "appId": "",
      "appPassword": "",
      "id": "24"
    }
  ],
  "padlock": "",
  "version": "2.0"
}

**Pull request merged**